pymdownx/blocks/block.py

```diff
@@ -205,10 +221,10 @@ class Block(metaclass=ABCMeta):
     NAME = ''
 
     # Instance arguments and options
-    ARGUMENT = False
-    OPTIONS = {}
+    ARGUMENT: bool | None = False
+    OPTIONS: dict[str, Any] = {}
```

Comment:
not sure if this should be more restrictive

Reply:
Can you elaborate? There seems to be some concern here, but I would need a bit more reasoning as to why there is concern here.

Reply:
Yeah, so it is a dict with a string key, but currently it is defined with a list of of two items: a default, and a function to parse the current input.

Obviously, list is a bad choice if typing. Tuple would be better because then we could be more specific: `dict[str, tuple[Any, Callable[[Any], _T]]` (or something like that).

Reply:
If we want to do something like this, we need to adjust existing extensions to use a Tuple and update documentation to showcase the tuple instead of lists as it currently does.

Reply:
I would welcome that change but it feels like it belongs in a separate PR

Reply:
I'm not sure I agree, the point of this PR is to provide good, accurate types for this submodule, and it should be a non-functional change.

Reply:
Maybe I'm misunderstanding, what is the change that you propose?

Reply:
Sorry, updated the last statement to specify the tuple should contain a default and a callback, not a string and callback.

Reply:
I see, I would usually split this kind of behaviour change into a separate PR from a type annotation change. How would you feel about me opening that PR first, merging that, and then coming back to this one to add the types?

Reply:
As the reviewer and the maintainer, I don't mind these changes being combined as these are related to getting this submodule properly typed, but if you want to do a separate PR, I won't stop you.

Reply:
Fair enough, I'll add it to this one within the next few days.

Reply:
I updated the type and the examples, please let me know if I've missed anything.